hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldOperations.java

```diff
@@ -182,6 +228,11 @@ public Object readObject(GenericRecord genericRecord, String fieldName) {
         public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord genericRecord, String fieldName) {
             writer.writeString(fieldName, genericRecord.getString(fieldName));
         }
+
+        @Override
+        public String readAsJsonFormattedField(AbstractGenericRecord record, String fieldName) {
+            return "\"" + record.getString(fieldName) + "\"";
```

Comment:
When it looks like JSON, it probably should be JSON and we should escape the strings. Also field names should be escaped then.

Reply:
Make sense. I wonder if we have anything reusable in our codebase. `JsonWriter` looks like have somethinng but not reusable at the moment.

Reply:
@viliam-durina Updated the pr accordingly. Can you check again ?

Reply:
I only glanced, looks ok. But it's quite inefficient. Much better way would be if the escape method directly copied to the StringBuilder, e.g. `public static void escape(String source, StringBuilder target)`